loadtest/control/simulcontroller/controller.go

```diff
@@ -88,15 +88,16 @@ func (c *SimulController) Run() {
 		},
 	}
 
-	for _, action := range initActions {
+	for i := 0; i < len(initActions); i++ {
 		select {
 		case <-c.stopChan:
 			return
 		case <-time.After(pickIdleTimeMs(c.config.MinIdleTimeMs, c.config.AvgIdleTimeMs, 1.0)):
 		}
 
-		if resp := action.run(c.user); resp.Err != nil {
+		if resp := initActions[i].run(c.user); resp.Err != nil {
 			c.status <- c.newErrorStatus(resp.Err)
+			i--
```

Comment:
Wondering if it's okay to unconditionally retry forever ? Would it make sense to add a cap to the number of retries ?

Reply:
This is similar to what we've done with login but applied to the whole set of init actions. These actions are required for the controller to run so either we re-try indefinitely or we exit (stop the controller) after we reach some sort of max number of failures (what you suggested). 
The problem with the latter approach is that, at the moment, we don't have a good way to notify the coordinator we stopped a controller. 

I think we might revisit this if we figure out it makes more sense to stop them directly from here rather than waiting for the coordinator to do it due to performance decreasing. Mainly trying to keep the logic simple even at the cost of being a bit dumb.

Reply:
A controller does not have any logic to stop itself after it starts to run AFAIR, this one seems OK to me.

Reply:
Sounds good then.